src/ui/components/ExplorerHome/ExplorerTimeline.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+import React, {type Node as ReactNode} from "react";
+import {extent} from "d3-array";
+import {scaleLinear} from "d3-scale";
+import {line} from "d3-shape";
+import {type Grain} from "../../../core/ledger/grain";
+
+const legendTextStyles = {
+  fontSize: "10px",
+  fill: "#fff",
+};
+const CRED_COLOR = "#6174CC";
+const GRAIN_COLOR = "#FFAA3D";
+type ExplorerTimelineProps = {|
+  +timelines: {
+    cred: Array<number>,
+    grain?: Array<Grain>,
+  },
+  +width?: number,
+  +height?: number,
+  +hasLegend?: boolean,
+  +responsive?: boolean,
+|};
+const ExplorerTimeline = (props: ExplorerTimelineProps): ReactNode => {
+  if (props.timelines.cred == null && props.timelines.grain == null) {
+    return <></>;
+  }
+  const grainExists = props.timelines.grain && true;
+  const width = props.width || 300;
+  const height = props.height || 25;
+  const viewBox = `0 0 ${width} ${height}`;
+  const intervals = props.timelines.cred.length;
+  let grainAsNumber, range, grainValues;
+  if (grainExists) {
+    const grain = props.timelines.grain || [];
+    grainAsNumber = grain.map((g) => {
```

Comment:
This would be better as a `const`

Reply:
looks like we need to leave `grainValues` in a `let` statement on line 33 but everything else works as a `const`